astropy/utils/tests/test_data.py

```diff
@@ -372,3 +376,37 @@ def test_is_url_in_cache():
 
     download_file(TESTURL, cache=True, show_progress=False)
     assert is_url_in_cache(TESTURL)
+
+
+@contextlib.contextmanager
+def TemporaryTempDir():
+    """Temporarily redirect all temporary files to a temporary directory"""
+    tempdir = tempfile.mkdtemp()
+    old_tempdir = tempfile.tempdir
+    tempfile.tempdir = tempdir
+    try:
+        yield tempdir
+    finally:
+        tempfile.tempdir = old_tempdir
+        shutil.rmtree(tempdir)
+
+
+def test_get_readable_fileobj_cleans_up_temporary_files():
+    """checks that get_readable_fileobj leaves no temporary files behind"""
+    from ..data import get_pkg_data_filename, get_readable_fileobj
```

Comment:
The in-line imports aren't necessary either.  You can just import these at the top of the file (if there are any tests already that do it, they're demonstrating an anti-pattern)